cinn/runtime/cpu/host_intrinsics.cc

```diff
@@ -134,6 +134,8 @@ inline int FN_INT32(clz)(int x) { return __builtin_clz(x); }
 
 inline int FN_INT32(popc)(int x) { return __builtin_popcount(x); }
 
+inline int FN_INT32(logical_right_shift)(int x, int y) { return (x >> y) & ~(((0x1 << 31) >> y) << 1); }
```

Comment:
逻辑右移的计算这么复杂的么。。。难道不是强转unsigned后右移就行？`((unsigned int)x >> y)`。请问这公式的来源是？

Reply:
现在这样确实就行了，之前没有区分cpu和nvgpu的实现，直接用ir::cast进行强转是不支持uint32的。

Reply:
咦？ir::Cast强转不支持uint32是指会报错么？还是生成的代码没有`(unsigned int)`这个强转标识符？如果是后者可以参考[`IrPrinter::Visit(const FloatImm *x)`](https://github.com/PaddlePaddle/CINN/blob/develop/cinn/ir/ir_printer.cc#L42)改下`IrPrinter::Visit(const UIntImm *x) `来保证最终生成的代码里强转标识符不会被优化掉。

退而求其次，也可以把这个直接改成：
```suggestion
inline int FN_INT32(logical_right_shift)(int x, int y) { return ((unsigned int)x >> y); }
```
感觉这样更清晰简洁。

Reply:
ir::Cast强转uint32会报错。`F1205 10:56:21.111316  3729 llvm_util.cc:74] Check failed: ir_type LLVM can't convert type: uint32`。ir::Cast强转uint32和uint64目前好像只支持立即数的强转。

Reply:
好的，我知道是啥问题了，这bug涉及的地方的确有点深，后续我来修复下吧。这个PR可以先合入